### PR TITLE
fix(core): keep chat-local asks out of CHARACTER gate

### LINKED ISSUE #18098: fix(core): CHARACTER_GATE_NOTICE refuses ordinary chat-local asks from low-role senders
## Post-merge deterministic regression

Merged PR #18091 added an `alwaysInResponseState` provider that treats the shared CHARACTER action heuristic as proof that a message requests a persistent character edit. The blocking review on `f48d32b6e75bc265883864cc6d7dc9d105a45eed` identified that `speechRule` and `degreeDirective` require neither persistent scope nor a character/configuration cue, but the PR was merged as `75c9a07bac949b1f1d8f05485cdbafe8bd2ccd37` without addressing it.

The relevant implementation and test blobs are byte-identical at PR head `f48d32b`, merge `75c9a07`, and current `develop@112a73c8eec7b6491208e6e04d3687482b905e62`.

On the exact current tree, a hermetic no-env probe under Node `24.15.0` / Bun `1.3.14` produces:

```text
never mention this code in this chat       definitive=true modification=true gated=true requiredRole=ADMIN ordersDecline=true
always use metric units in this answer      definitive=true modification=true gated=true requiredRole=ADMIN ordersDecline=true
be more careful with this transaction       definitive=true modification=true gated=true requiredRole=ADMIN ordersDecline=true
```

For a connector sender resolved as GUEST, each ordinary turn-local request therefore injects `# Character modification access notice` and the strong instruction `Do not promise... Acknowledge ... that only your owner/admins can change how you are configured.` The final caveat suggesting the model reinterpret chat-local scope comes only after the provider has deterministically asserted a persistent edit and ordered a decline.

The two focused suites remain green (`22/22`), but `character-gate-notice.test.ts` contains no low-role negative coverage for these shapes, so the regression is outside the asserted contract.

## Impact

Low-role Discord/connector users can receive an authorization refusal for normal conversation-scoped safety, formatting, or caution requests that require no persistent character mutation.

## Acceptance criteria

- Do not use the broad action fast path as an always-on authorization signal without a persistent character/configuration cue or an equivalent disambiguation boundary.
- Preserve the CHARACTER action's fast path once a request has actually been routed into CHARACTER; this issue is about global provider reuse.
- Add GUEST negative provider tests for all three exact messages above; each must return an empty notice and no `characterModificationGated` value.
- Keep a positive low-role test for an explicit persistent request such as `change your personality to never say bet`, including effective `ACTION_ROLE_POLICY` handling.
- Prove the focused provider/action suites green under the repository-pinned Node/Bun versions.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Codex
Contribution skill revision: N/A - contribute-to-eliza skill was not available
Attribution status: self-reported
— [codex]

### PR DESCRIPTION
**Pull request by NubsCarson**
# Relates to

Closes #18098.

- [x] This PR targets `develop` and is rebased onto exact `origin/develop@6659f25cd20eea80e4ad4a8a61a1e541b21551df` with zero conflicts.
- [x] `bun install` and `bun run verify` were run after sync.
- [x] A reviewer can verify the behavior from the attached trajectory artifacts and the explicit local/persistent matrix below.

# Contribution provenance

- AI assistance: `yes`

- Model(s) used: `openai/gpt-5.6-sol`

- Agent tooling: `Codex`

- Skill path: `N/A - no contribution skill used`

- Provenance status: `self-reported`

# Sync with develop

- Exact head: `ae71d900916d26dbbea8654b6c7c8a726872f3b2`
- Tree: `2ff6f9962253f7872ca0b8fd8b789083b99d3386`
- Parent/base: `6659f25cd20eea80e4ad4a8a61a1e541b21551df`
- [x] Rebased onto the latest `origin/develop`; zero conflicts.
- [x] `bun run verify` passes post-sync: 346/346 Turbo tasks plus every repository audit.

# Risks

Medium. This changes a high-precision authorization notice classifier. A false positive refuses a harmless turn-local request; a false negative omits the explanatory notice for a persistent CHARACTER request. The provider remains deliberately bounded, has 232 hardened-connector/local/persistent/coordination/policy regressions, and received an independent exact-object review with no P0/P1/P2 findings.

# Background

## What does this PR do?

- Replaces the global provider's reuse of the broad routed-action heuristic with a bounded high-confidence persistent-request classifier.
- Treats reply-, response-, message-, thread-, chat-, task-, transaction-, and other explicit occasion scopes as turn-local while preserving standing complements and explicit persistence.
- Keeps coordination, negation, quotation, named-address, polite-prefix/suffix, and `ACTION_ROLE_POLICY` precedence clause-scoped.
- Reads the canonical user payload after real connector hardening instead of classifying the generated external-content armor.
- Leaves the CHARACTER action's routed fast path intact for authorized persistent edits.

## What kind of change is this?

Bug fix (non-breaking).

# Documentation changes needed?

No user-facing documentation change is required; the change restores the documented role-gate behavior and freezes it in provider/action tests.

# Testing

## Where should a reviewer start?

1. `packages/core/src/features/advanced-capabilities/personality/__tests__/character-gate-notice.test.ts`
2. `packages/core/src/features/advanced-capabilities/personality/providers/character-gate-notice.ts`
3. The attached report/native trace/viewer: rows 1-3 are harmless local asks with no gate; row 4 is an explicit persistent request with the ADMIN notice; the character config is unchanged.

## Detailed testing steps

- `bun test packages/core/src/features/advanced-capabilities/personality/__tests__/character-gate-notice.test.ts packages/core/src/features/advanced-capabilities/personality/__tests__/character-modify-intent.test.ts` — 232/232.
- `bun run --cwd packages/core test` — 501 files, 5,186 passed, 1 existing skip.
- `bun run --cwd packages/core typecheck` — pass.
- `bun run --cwd packages/core build` — Node, browser, edge, testing and declarations pass.
- `bunx biome check <exact three paths>` and `git diff --check` — pass.
- `RUN_TURBO_CONCURRENCY=4 bun run verify` — 346/346 plus all audits.

# Evidence Gate

- [ ] N/A - Core provider behavior has no rendered UI surface.

- [ ] N/A - Core provider behavior has no rendered UI surface.

- [ ] N/A - no visual interaction or UI state changes.

- [ ] N/A - no HTTP/backend route changes; the combined native integration trace is linked under trajectory evidence.

- [ ] N/A - no frontend code, console, or network surface changes.

- [x] [18163-character-gate-llm-trajectory-18163.json](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-character-gate-llm-trajectory-18163.json)

- [x] [18163-matrix.json](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-matrix.json)

- [ ] N/A - no rendered text or UI surface.

# Evidence Details

## Real LLM-call trajectory

The run uses the live Codex subscription model `openai/gpt-5.6-sol` through a real `AgentRuntime` and PGLite path. The scenario report labels its execution profile `simulated`, so this is supplemental diagnostic evidence rather than a claim of framework-qualified provider evidence.

- [Machine-verifiable redacted LLM calls](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-character-gate-llm-trajectory-18163.json)
- [Scenario report](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-report.json)
- [Native redacted JSONL](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-native.jsonl)
- [Run viewer bundle](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-character-gate-viewer-18163.zip)
- [Per-call trajectory bundle](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-character-gate-trajectories-18163.zip)
- [Privacy attestation](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-native.privacy-attestation.json)

Observed matrix:

1. local code discretion request → normal reply, no CHARACTER gate/mutation;
2. local metric-units request → normal reply, no CHARACTER gate/mutation;
3. transaction-local caution request → normal reply, no CHARACTER gate/mutation;
4. persistent personality request → ADMIN authorization notice, no mutation.

The final character configuration matches the starting configuration.

## Backend + frontend logs

N/A - no server or frontend surface changed. The [native scenario JSONL](https://github.com/elizaOS/eliza/releases/download/pr-evidence-7/18163-native.jsonl) records the real model requests/responses, reply selection, and final state.

Artifact SHA-256 values:

- trajectory document: `64cb1efd7e4a6586cd20f3563d2a239fed2026e7ca359100ad35654b42c08e09`
- native JSONL: `48c322476979c02533236e83447b439a4ef93fee2e621c5eeb0ccff7d59c8cb0`
- scenario report/matrix: `899396a7fe8c2fd845aabcd60689c41ecb44cdd812bb86bc2d39131d8c436f5e`

## Screenshots (before / after) + video walkthrough

N/A - Core-only provider/action behavior with no rendered UI.

## Audio / voice walkthrough

N/A - no voice, transcript, TTS, or STT behavior changed.

## Known gaps / failures

- The live-model diagnostic report is explicitly marked `executionProfile: simulated`; it supplements but does not replace the 232-case hardened-connector matrix, full Core suite, build, and root verification.
- Land #18162 before this PR. Without that trust-boundary repair, Stage 1 can misclassify generated connector armor before this provider sees the canonical payload.
- No UI/backend/audio artifacts apply because this patch changes only Core prompt-state/provider classification and its tests.

# Additional scope (commit 58c6915770)

## Disengagement directives must not gate

Live incident (busy multi-user channel, 2026-08-09): senders asked the agent to stop replying, the standing-rule classifier read those turns as unauthorized persistent character-modification requests, and the gate notice scripted permissions retorts instead of letting the turn disengage. STOP was emitted zero times in the window because the notice preempted the normal response path.

Structural fix at the same classification seam this PR already narrows — no post-filtering of model output:

- A standing-rule match whose content is a cessation or quieting predicate ("stop replying", "be quiet", "never respond to everything") with only audience/extent/persistence tail markers is classified as a conversation-scoped engagement request and never gates; the turn flows to normal handling where STOP/IGNORE is reachable.
- Content-bearing tails keep gating ("never reply with your real name", "never say bet again"), as do genuine configuration mutations; `rewrite your (system) prompt` is now recognized as one.
- Boundary decision: "stop being so rude" is chat-local tone feedback and stays ungated (no persistence marker, answerable in-turn without configuration change); "Never be rude from now on." remains gated. Both sides are pinned by tests.
- Explicitly NOT implemented: any persistent mute or role state — a low-role sender still cannot silence the agent beyond what the normal response path chooses for the turn.

Tests: +27 disengagement phrasings (16 fail against the pre-commit classifier), +6 still-gated persona-modification asks, +1 boundary case — `character-gate-notice.test.ts` 243/243, full personality suite 328/328, `bun run --cwd packages/core typecheck` pass. Changed-file coverage from the changed tests: 97.3% statements / 99.3% lines on the provider.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Codex
Contribution skill revision: N/A - no contribution skill used
Attribution status: self-reported
— [codex]

### CODE REVIEW

**Comment by standujar** [comment]
CLAIMING REVIEW: exact head ae71d900916d26dbbea8654b6c7c8a726872f3b2 against live develop bf030f0b66ea06ab8f553fef4331b64af1870145. Scope: issue acceptance criteria, current-base integration, classifier architecture, security/privacy and failure semantics, adversarial/concurrency-relevant cases, tests, checks, evidence, and contribution provenance. I will not merge or mutate the author branch.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Codex
Contribution skill revision: elizaOS/eliza@6659f25cd20eea80e4ad4a8a61a1e541b21551df:packages/skills/skills/contribute-to-eliza
Attribution status: self-reported
— [codex]

File: packages/core/src/features/advanced-capabilities/personality/providers/character-gate-notice.ts [hunk 1]
```diff
@@ -1,41 +1,315 @@
...
+	const addressCandidate = text.trim().replace(LEADING_REQUEST_WORD, "").trim();
+	const namedAddressMatch = addressCandidate.match(UNKNOWN_NAMED_ADDRESS);
+	const namedAddress = namedAddressMatch?.[1] ?? namedAddressMatch?.[2];
+	if (
+		namedAddress &&
+		namedAddress.toLocaleLowerCase() !== agentName?.trim().toLocaleLowerCase()
+	) {
+		return false;
+	}
+	return requestClauses(text, agentName).some((clause) =>
```
Comment (standujar) [inline packages/core/src/features/advanced-capabilities/personality/providers/character-gate-notice.ts:227]: [P1] The new high-confidence boundary still reverses both sides of the authorization decision for clear inputs. On this exact head, a pinned no-network provider test returns `gated=true` for five explicit one-turn, quoted, or informational requests: `Change your tone to formal for this sentence.`, `Always use metric units for the next calculation.`, `Reply with the exact words "change your personality permanently."`, `Answer this question: should I change your personality permanently?`, and `Never mention spoilers while discussing this book.` It returns `gated=false` for five unequivocal persistent asks: `Thanks. Change your personality permanently.`, `What do you think? Going forward, always respond formally.`, `Actually, change your personality permanently.`, `Stop saying bet forever.`, and `Make your personality more formal forever.` The first set still gets the false ADMIN refusal this PR is meant to remove; the second set still lets a low-role turn proceed without the refusal signal. Please replace or broaden the finite artifact/verb grammar and leading-clause shortcut so these semantics are stable, then add all ten as provider regressions.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Codex
Contribution skill revision: elizaOS/eliza@6659f25cd20eea80e4ad4a8a61a1e541b21551df:packages/skills/skills/contribute-to-eliza
Attribution status: self-reported
— [codex]

**Review by standujar — CHANGES_REQUESTED** [review]
Reviewed exact head ae71d900916d26dbbea8654b6c7c8a726872f3b2 against live develop bf030f0b66ea06ab8f553fef4331b64af1870145. I am requesting changes for the correctness blocker on the classifier line and for the evidence gap below.

Evidence blocker: the attached matrix reports `executionProfile: simulated`, `qualificationCounts.ineligible: 1`, `qualified: 0`, and `publishableScenarioCount: 0`; the trajectory labels itself `diagnostic-only`. The first three local-request turns also answer only “How can I help?”, so the artifact proves absence of the role notice but not that the local instruction is honored. Core provider→model changes require a qualified live-model trajectory. Please recapture a qualified exact-head run after the classifier repair, with assertions for both no false refusal and a natural response that follows each local instruction.

Independent verification:
- current-base synthetic merge tree `74b1188f3a2965a21941f477a7bd70bc076cb36f`; original and rebased patch-id both `0c67f8679bedd6444b7677cef2892dbfa749788b`;
- changed executable/control surfaces: none; only three regular TypeScript blobs;
- pinned Bun 1.3.14, no-network/read-only container: PR suites 232/232 pass;
- independent adversarial matrix: 0/10 pass, with all ten observed decisions inverted from the clear local/persistent semantics listed inline;
- attached artifact SHA-256 values match the PR body and the privacy attestation passes;
- CI is green except `Tests` and `Smoke`, which are still pending at review time.

AI provider/model: OpenAI / gpt-5.6-sol
Client / agent tooling: Codex
Contribution skill revision: elizaOS/eliza@6659f25cd20eea80e4ad4a8a61a1e541b21551df:packages/skills/skills/contribute-to-eliza
Attribution status: self-reported
— [codex]

**Comment by NubsCarson** [comment]
Red `Tests`/`Smoke` lanes here are develop-wide CI breakage, not this change — repair is #18193 (plugin-workflow wedge, retired `--min-tasks` flag, lane budget). Lanes will be rerun once it lands.
